lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -140,13 +172,35 @@ def configure(conf)
       if @cache_ttl < 0
         @cache_ttl = :none
       end
-      @cache = LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl)
+      @pod_cache = LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl)
+      @pod_cache_watch_updates = 0
```

Comment:
Any reason we cant make this string of attributes into a hash? and initialize them all.  This is just a bunch of counters/metrics right?  We could create a simple 'counter' class

```
class Counter
  def initialize(name)
     @name = name
     @watch_misses = 0
     ....
  end
  attr_reader :name
  attr_reader :watch_misses

  def watch_missed
    @watch_misses += 1
  end
  ....
end

pod_cache_watch_counter = Counter.new('pod_cache')
pod_cache_watch_counter.watch_missed

Reply:
https://github.com/balgarath/ruby-counter-class